syntax = "proto3";

package character.v1;

service CharacterService {
  rpc GetCharacterReward (RewardRequest) returns (RewardResponse) {}
  rpc GetDefaultCharacter (GetDefaultCharacterRequest) returns (GetDefaultCharacterResponse) {}

  // Chat Worker용: 매칭 라벨로 캐릭터 조회
  rpc GetCharacterByMatch (GetByMatchRequest) returns (GetByMatchResponse) {}
}

// Chat Worker용 메시지
message GetByMatchRequest {
  string match_label = 1;  // 폐기물 중분류 (예: "플라스틱", "종이류")
}

message GetByMatchResponse {
  bool found = 1;
  string character_name = 2;
  string character_type = 3;
  string character_dialog = 4;
  string match_label = 5;
}

message GetDefaultCharacterRequest {}

message GetDefaultCharacterResponse {
  bool found = 1;
  string character_id = 2;
  string character_code = 3;
  string character_name = 4;
  string character_type = 5;
  string character_dialog = 6;
}

message ClassificationSummary {
  string major_category = 1;
  string middle_category = 2;
  optional string minor_category = 3;
}

message RewardRequest {
  string source = 1; // enum: scan
  string user_id = 2; // UUID string
  string task_id = 3;
  ClassificationSummary classification = 4;
  repeated string situation_tags = 5;
  bool disposal_rules_present = 6;
  bool insufficiencies_present = 7;
}

message RewardResponse {
  bool received = 1;
  bool already_owned = 2;
  optional string name = 3;
  optional string dialog = 4;
  optional string match_reason = 5;
  optional string character_type = 6;
  optional string type = 7;
}
